components/script/layout_wrapper.rs

```diff
@@ -824,6 +801,14 @@ impl<'ln> ThreadSafeLayoutNode for ServoThreadSafeLayoutNode<'ln> {
         self.node.type_id()
     }
 
+    fn style_for_text_node(&self) -> Arc<ComputedValues> {
```

Comment:
Hm... Could we cache this somehow? This will become a hot path now, `fn style(&self, &SharedLayoutContext)` is called a zillion times from layout, assuming it's super-cheap.